lib/ruby_terraform/command_line/options.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'option'
+require_relative 'flag'
+require_relative 'boolean_option'
+
+module RubyTerraform
+  module CommandLine
+    class Options < Array
+      def initialize(option_values:, command_arguments:)
```

Comment:
Please add `nil`s.

Reply:
Does this class represent both options and arguments? If so, I'm not sure the name is clear enough. I wonder if there's a name that is more generic that covers both of those things. Maybe `Parts`? I'm not sure what term is used by, say, unix to refer to the parts of a command invocation. Maybe it is 'options' since you have things like `optparse`, `OptionParser` in Ruby. However in that case, I feel that the `Option` class should maybe be called `Switch` since it specifically refers to a `--thing=val` option or `["--thing", "val"]` options. Then we'd have `Options` representing the collection of `Switch`, `Flag` and `Argument`.

I think `lino` might also mix up this language so would be good to come up with something definitive.

Reply:
This only represents the command line options (-option=xyz) and command line flags (-option).  The arguments were just strings so didn't need all the complexity that these did.. 

In terms of nils - 

- if the option_values are nil (ie the user hasn't passed in any parameters to the RubyTerraform command, then I'd want this to go bang! I think (ie default nil would be pointless as no flags or options will be created in the array.
- if the command arguments are nil, then the Command hasn't passed in what values it wants extracted the option_values, so again we'd be loading nothing into the array of Options and it should go bang.  

(or am I misinterpreting what you're asking for here?)